SC-18599 add STAW support (#609)
sematext/docs: docs/agents/sematext-agent/windows-installation.md

## Install from GUI

1. Download the [Sematext Agent installer](https://todo-add-public-installer-repo)
2. Run the installer as an Administrator
3. Follow the installation steps and accept the license agreement 
4. Set the Infra App token
5. Select a region (EU or US) based on where you created your account in Sematext Cloud

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

## Install from command line

Download the [Sematext Agent installer](https://todo-add-public-installer-repo).

### Using PowerShell

Run the following command as Administrator from the folder that you downloaded the installer:

```
    Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'sematext-agent-VERSION.msi'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=YOUR_INFRA_TOKEN REGION=YOUR_REGION_EU_or_US"
```

> Make sure to replace `YOUR_INFRA_APP_TOKEN` with your Infra App token and `YOUR_REGION_EU_or_US` with your region. For example:

Reviewer: You can mention file name also, `VERSION`

```
    Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'sematext-agent-3.3.0.2.msi'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=7511db7f-c060-4e10-b667-5f2653d4933e REGION=EU"
```

### Using Windows CMD:

Run the following command as Administrator from the folder that you downloaded the installer:

```
    start /wait msiexec /qn /i "sematext-agent-VERSION.msi" /L*V "sematext-agent-VERSION.msi.log" INFRA_TOKEN=YOUR_INFRA_APP_TOKEN REGION=YOUR_REGION_EU_or_US
```

Make sure to replace `YOUR_INFRA_APP_TOKEN` with your Infra App token and `YOUR_REGION_EU_or_US` with your region. For example:

```
    start /wait msiexec /qn /i "sematext-agent-3.3.0.2.msi" /L*V "sematext-agent-3.3.0.2.msi.log" INFRA_APP_TOKEN=7511db7f-c060-4e10-b667-5f2653d4933e REGION=EU
```

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

## How to check the Sematext Agent service status

### Windows Services

Open Windows Services and find the Sematext Agent service:

![Sematext Agent service](images/sematext-agent-service.png)

### Command line

Run the following command:

```

> cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service status

Reviewer: This is not a inline command, make it inline or 2 lines

```

## How to start / stop / restart Sematext Agent

Run one of the following commands to start, stop or restart Sematext Agent

Start:

```

> cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service start

Reviewer: This is not a inline command, make it inline or 2 lines

```

Stop:

```

> cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service stop

Reviewer: This is not a inline command, make it inline or 2 lines

```

Restart:

```

> cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service restart

Reviewer: This is not a inline command, make it inline or 2 lines

```

## How to check Sematext Agent logs

> Sematext Agent logs are located in the `%LOCALAPPDATA%\sematext-agent\logs\` folder.

Reviewer: Let's say this folder: `C:\Windows\system32\config\systemprofile\AppData\Local\sematext-agent`

And you may want to mention this example;

```
cd C:\Windows\system32\config\systemprofile\AppData\Local\sematext-agent

notepad.exe .\logs\st-agent.log
```

## How to change Infra App token and region

You can either reinstall Sematext Agent through GUI or by running the following command from command line:

```
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-config set-token -t "YOUR_INFRA_TOKEN"

> .\st-agent-amd64.exe windows-config set-region -r "YOUR_REGION_EU_or_US"

Reviewer: Also mention restart;

After the changes, you need to restart the agent.
```
.\st-agent-amd64.exe windows-service restart
```

```

## How to unistall Sematext Agent

> You can uninstall Sematext Agent using the GUI installer or by from Windows Settings > Apps

Reviewer: ```suggestion
You can uninstall Sematext Agent using the GUI installer or by from Windows Settings > Apps.
```

## Limitations

The Sematext Agent for Windows is still a work in progress and has some limitations:

- Discovery and configuration of services and log sources are not yet supported
- Inventory functionality is not yet available

> - CPU and memory consumption of Sematext Agent are not yet optimized

Reviewer: Let's remove this one

> - Custom tags cannot be created

Reviewer: Not sure about this one.

Author: I will leave it as is for now

```
start /wait msiexec /qn /i "sematext-agent-3.3.0.2.msi" /L*V "sematext-agent-3.3.0.2.msi.log" INFRA_APP_TOKEN=7511db7f-c060-4e10-b667-5f2653d4933e REGION=EU
```

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

## How to check the Sematext Agent service status

### Windows Services

Open Windows Services and find the Sematext Agent service:

![Sematext Agent service](images/sematext-agent-service.png)

### Command line

Run the following command:

```
cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service status
```

## How to start / stop / restart Sematext Agent

Run one of the following commands to start, stop or restart Sematext Agent

Start:

```
cd 'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-service start
```

Stop:

```
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-service stop
```

Restart:

```
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-service restart
```

## How to check Sematext Agent logs

Sematext Agent logs are located in the `C:\Windows\system32\config\systemprofile\AppData\Local\sematext-agent` folder. For example:

```

> cd C:\Windows\system32\config\systemprofile\AppData\Local\sematext-agent

Reviewer: Let's update the path like this `C:\Windows\System32\config\systemprofile\AppData\Local\sematext-agent`

Installation can be done either by using the GUI installer or through the command line.

## Pre-requirements

Before starting, please check the [platform support policy](https://sematext.com/docs/agents/sematext-agent/platform-support-policy/) page to make sure your Windows platform is supported.

To see your Windows host metrics in Sematext Cloud, you need to [create an Infra App](https://sematext.com/docs/monitoring/infrastructure/#create-an-infra-app) and select the Windows environment. This will provide you with an Infra App token, which is essential for establishing a secure link between Sematext Cloud and Sematext Agent.

## Install from GUI

1. Download the [Sematext Agent installer](https://todo-add-public-installer-repo)
2. Run the installer as an Administrator
3. Follow the installation steps and accept the license agreement 
4. Set the Infra App token
5. Select a region (EU or US) based on where you created your account in Sematext Cloud

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

> 

Reviewer: ```suggestion
```

## Install from command line

Download the [Sematext Agent installer](https://todo-add-public-installer-repo).

### Using PowerShell

Run the following command as Administrator from the folder that you downloaded the installer:

```
    Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'sematext-agent-VERSION.msi'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=YOUR_INFRA_TOKEN REGION=YOUR_REGION_EU_or_US"
```

Where:

- `YOUR_INFRA_APP_TOKEN`: the Infra App token
- `YOUR_REGION_EU_or_US`: your Sematext Cloud region
- `VERSION`: the agent version

For example:

```

>     Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'sematext-agent-3.3.0.2.msi'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=7511db7f-c060-4e10-b667-5f2653d4933e REGION=EU"

Reviewer: ```suggestion
Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'sematext-agent-3.3.0.2.msi'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=7511db7f-c060-4e10-b667-5f2653d4933e REGION=EU"
```

```

### Using Windows CMD:

Run the following command as Administrator from the folder that you downloaded the installer:

```

>     start /wait msiexec /qn /i "sematext-agent-VERSION.msi" /L*V "sematext-agent-VERSION.msi.log" INFRA_TOKEN=YOUR_INFRA_APP_TOKEN REGION=YOUR_REGION_EU_or_US

Reviewer: ```suggestion
start /wait msiexec /qn /i "sematext-agent-VERSION.msi" /L*V "sematext-agent-VERSION.msi.log" INFRA_TOKEN=YOUR_INFRA_APP_TOKEN REGION=YOUR_REGION_EU_or_US
```

```

Where:

- `YOUR_INFRA_APP_TOKEN`: the Infra App token
- `YOUR_REGION_EU_or_US`: your Sematext Cloud region
- `VERSION`: the agent version

For example:

```
    start /wait msiexec /qn /i "sematext-agent-3.3.0.2.msi" /L*V "sematext-agent-3.3.0.2.msi.log" INFRA_APP_TOKEN=7511db7f-c060-4e10-b667-5f2653d4933e REGION=EU
```

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

## How to check the Sematext Agent service status

### Windows Services

Open Windows Services and find the Sematext Agent service:

![Sematext Agent service](images/sematext-agent-service.png)

### Command line

Run the following command:

```

> cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service status

Reviewer: ```suggestion
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-service status
```

```

## How to start / stop / restart Sematext Agent

> Run one of the following commands to start, stop or restart Sematext Agent

Reviewer: You may want to mention that it's possible to do that via GUI (Windows Services application)

Start:

```
cd 'C:\Program Files\Sematext Agent\'

> .\st-agent-amd64.exe windows-service status

Reviewer: ```suggestion
.\st-agent-amd64.exe windows-service start
```

```

Stop:

```
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-service stop
```

Restart:

```
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-service restart
```

## How to check Sematext Agent logs

> Sematext Agent logs are located in the `C:\Windows\system32\config\systemprofile\AppData\Local\sematext-agent` folder. For example:

Reviewer: ```suggestion
Sematext Agent logs are located in the `C:\Windows\system32\config\systemprofile\AppData\Local\sematext-agent` folder.

For example, you can access the logs file like below;
```

```
cd C:\Windows\system32\config\systemprofile\AppData\Local\sematext-agent

> notepad.exe .\logs\st-agent.log

Reviewer: ```suggestion

notepad.exe .\logs\st-agent.log
```

```

## How to change Infra App token and region

You can either reinstall Sematext Agent through GUI or by running the following command from command line:

```
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-config set-token -t "YOUR_INFRA_TOKEN"
.\st-agent-amd64.exe windows-config set-region -r "YOUR_REGION_EU_or_US"

> .\st-agent-amd64.exe windows-service restart

Reviewer: You can either reinstall Sematext Agent through GUI or by running the following command from command line:

```
cd  'C:\Program Files\Sematext Agent\'
.\st-agent-amd64.exe windows-config set-token -t "YOUR_INFRA_TOKEN"
.\st-agent-amd64.exe windows-config set-region -r "YOUR_REGION_EU_or_US"
```

After the changes, you need to restart the agent:
```
.\st-agent-amd64.exe windows-service restart
```

```

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

## How to check the Sematext Agent service status

### Windows Services

Open Windows Services and find the Sematext Agent service:

![Sematext Agent service](images/sematext-agent-service.png)

### Command line

Run the following command:

```
cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service status
```

## How to start / stop / restart Sematext Agent

To start, stop or restart the Sematext Agent, you can perform these actions through the GUI:

- Start: Open the Windows Services application by searching for "Services" in the Windows search bar, locate "Sematext Agent" in the list of services, right-click on it, and select "Start"
- Stop: Similarly, in the Services application, locate "Sematext Agent", right-click on it, and select "Stop"
- Restart: To restart the Sematext Agent, you can either stop and then start it again using the aforementioned steps or right-click on "Sematext Agent" in the Services application and select "Restart"

> Alternatively, you can use the command-line interface by running one of the following commands:

Reviewer: ```suggestion
Alternatively, you can use the command-line interface by running one of the following commands as Administrator:
```

title: Installing Sematext Agent on Windows

Sematext Agent collects a plethora of metrics about Windows hosts (cpu, memory, disk, network, processes). The installer is responsible to install the Sematext Agent, configuring it, and starting it as a Windows service.

Installation can be done either by using the GUI installer or through the command line.

## Pre-requirements

Before starting, please check the [platform support policy](https://sematext.com/docs/agents/sematext-agent/platform-support-policy/) page to make sure your Windows platform is supported.

To see your Windows host metrics in Sematext Cloud, you need to [create an Infra App](https://sematext.com/docs/monitoring/infrastructure/#create-an-infra-app) and select the Windows environment. This will provide you with an Infra App token, which is essential for establishing a secure link between Sematext Cloud and Sematext Agent.

## Install from GUI

1. Download the [Sematext Agent installer](https://todo-add-public-installer-repo)

> 2. Run the installer as an Administrator

Reviewer: I guess no need to this step. I have tested with Windows 10, It worked fine. I asked for admin permession when it's needed. So we can remove this step, I think.

See the Install button;
![Screenshot from 2024-04-24 18-34-17](https://github.com/sematext/docs/assets/20258973/b0669cf7-26f8-4091-b97d-725cf80e625a)

3. Follow the installation steps and accept the license agreement 
4. Set the Infra App token
5. Select a region (EU or US) based on where you created your account in Sematext Cloud

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

## Install from command line

Download the [Sematext Agent installer](https://todo-add-public-installer-repo).

### Using PowerShell

Run the following command as Administrator from the folder that you downloaded the installer:

```

> Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'sematext-agent-VERSION.msi'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=YOUR_INFRA_TOKEN REGION=YOUR_REGION_EU_or_US"

Reviewer: ```suggestion
Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'FILENAME'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=YOUR_INFRA_TOKEN REGION=YOUR_REGION_EU_or_US"
```

```

Where:

- `YOUR_INFRA_APP_TOKEN`: the Infra App token
- `YOUR_REGION_EU_or_US`: your Sematext Cloud region

> - `VERSION`: the agent version

Reviewer: ```suggestion
- `FILENAME`: blabla
```

For example:

```
Start-Process -Wait msiexec -ArgumentList "/qn /i $($msiFileName = 'sematext-agent-3.3.0.2.msi'; $msiFileName) /L*V `"$msiFileName.log`" INFRA_APP_TOKEN=7511db7f-c060-4e10-b667-5f2653d4933e REGION=EU"
```

### Using Windows CMD:

Run the following command as Administrator from the folder that you downloaded the installer:

```

> start /wait msiexec /qn /i "sematext-agent-VERSION.msi" /L*V "sematext-agent-VERSION.msi.log" INFRA_TOKEN=YOUR_INFRA_APP_TOKEN REGION=YOUR_REGION_EU_or_US

Reviewer: ```suggestion
start /wait msiexec /qn /i "FILENAME" /L*V "FILENAME.log" INFRA_TOKEN=YOUR_INFRA_APP_TOKEN REGION=YOUR_REGION_EU_or_US
```

```

Where:

- `YOUR_INFRA_APP_TOKEN`: the Infra App token
- `YOUR_REGION_EU_or_US`: your Sematext Cloud region

> - `VERSION`: the agent version

Reviewer: ```suggestion
- `FILENAME`: blabla
```

```

After completing these steps, Sematext Agent will be automatically started as a Windows Service.

## How to check the Sematext Agent service status

### Windows Services

Open Windows Services and find the Sematext Agent service:

![Sematext Agent service](images/sematext-agent-service.png)

### Command line

Run the following command:

```
cd  'C:\Program Files\Sematext Agent\' .\st-agent-amd64.exe windows-service status
```

## How to start / stop / restart Sematext Agent

To start, stop or restart the Sematext Agent, you can perform these actions through the GUI:

- Start: Open the Windows Services application by searching for "Services" in the Windows search bar, locate "Sematext Agent" in the list of services, right-click on it, and select "Start"
- Stop: Similarly, in the Services application, locate "Sematext Agent", right-click on it, and select "Stop"
- Restart: To restart the Sematext Agent, you can either stop and then start it again using the aforementioned steps or right-click on "Sematext Agent" in the Services application and select "Restart"

Alternatively, you can use the command-line interface by running one of the following commands:

Start:

```

> cd 'C:\Program Files\Sematext Agent\'

Reviewer: Looks like All Windows Service commands require Admin perm. See, I can't even run `status` command.

![Screenshot from 2024-04-24 18-36-28](https://github.com/sematext/docs/assets/20258973/85d1761c-553e-4ff2-8f01-63cd144ce1cc)

But I can run it if I have admin perms. 
![Screenshot from 2024-04-24 18-37-37](https://github.com/sematext/docs/assets/20258973/45474c91-4c9f-483e-b455-e422b978bc2e)

So they need to run their PowerShell with admin perms. I guess they would know "how-to".

Author: I think it should be ok. We can add `-Verb RunAs` at the end of each command but it may be too much.

Author: updated description
